### DISCUSSION TITLE
Need help to repair an old symfony 2.2 application with php version incompatibility

### DISCUSSION THREAD

**Discussion by inolil** [symfony/symfony#46778] [Q&A]
Hi,
In a previous life, I was a php developer. I loved Symfony and I developed many projects using this framework from its first version.
Now, I have completely different activities but I need an old symfony app for my current work.
This app is based on Symfony 2.2 and is hosted on a PHP 7.3 server.

The host provider warns me about my PHP version that will be vulnerable soon.

Could somebody help me to find the easiest way to run my app again, with the less code modifications ?

Now, app_dev.php gives me the following exception, probably due to incompatibility between Symfony and PHP versions :
`Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in /home/clients/web/neorural/task_manager/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php line 2511`

Thank you for your help

**Comment by ahundiak** [comment]
2.2?  That brings back some memories.  I do remember encountering the 'continue 2' error.  You need to update the doctrine/orm package to at least 2.6.6.  Assuming you trust your backups you can try `composer update doctrine/orm` and cross your fingers.  If it works then you should be past this particular error.  You will probably encounter more errors but at least they will be different errors.

In the far more likely event that composer blows up then I'm not sure what to do with something this old.  Refactoring to a newer Symfony version is the obvious choice especially if you need the app for the foreseeable future but that would probably take some time.

**Comment by inolil** [comment]
Thank you ahundiak,
After backed up, I tried to update doctrine but there was a lock file issue. I deleted composer.lock and tried again with `composer update` and get many errors.
Some packages are obsolete and there are some php errors.

Could you lead me to the next step ?

here is the log :


`uid113840@h2web139:~/web/neorural/task_manager$ composer update --no-dev doctrine/orm

                                                                                                                                                                  
  [RuntimeException]                                                                                                                                              
  The lock file does not contain require-dev information, run install with the --no-dev option or delete it and run composer update to generate a new lock file.  
                                                                                                                                                                  

update [--with WITH] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-autoloader] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...

uid113840@h2web139:~/web/neorural/task_manager$ composer update doctrine/orm
Cannot update only a partial set of packages without a lock file present. Run `composer update` to generate a lock file.
uid113840@h2web139:~/web/neorural/task_manager$ composer update
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Lock file operations: 35 installs, 0 updates, 0 removals
  - Locking doctrine/annotations (1.13.2)
  - Locking doctrine/cache (1.13.0)
  - Locking doctrine/collections (1.6.8)
  - Locking doctrine/common (2.13.3)
  - Locking doctrine/dbal (v2.4.5)
  - Locking doctrine/doctrine-bundle (v1.2.0)
  - Locking doctrine/event-manager (1.1.1)
  - Locking doctrine/inflector (1.4.4)
  - Locking doctrine/lexer (1.2.3)
  - Locking doctrine/orm (v2.4.8)
  - Locking doctrine/persistence (1.3.8)
  - Locking doctrine/reflection (1.2.3)
  - Locking jdorn/sql-formatter (v1.2.17)
  - Locking jms/aop-bundle (1.1.0)
  - Locking jms/cg (1.2.0)
  - Locking jms/di-extra-bundle (1.3.0)
  - Locking jms/metadata (1.7.0)
  - Locking jms/parser-lib (1.0.1)
  - Locking jms/security-extra-bundle (1.4.0)
  - Locking kriswallsmith/assetic (v1.1.3)
  - Locking monolog/monolog (1.27.1)
  - Locking phpoption/phpoption (1.8.1)
  - Locking psr/cache (3.0.0)
  - Locking psr/log (1.1.4)
  - Locking sensio/distribution-bundle (v2.2.6)
  - Locking sensio/framework-extra-bundle (v2.2.6)
  - Locking sensio/generator-bundle (v2.2.6)
  - Locking swiftmailer/swiftmailer (v5.0.3)
  - Locking symfony/assetic-bundle (v2.1.3)
  - Locking symfony/monolog-bundle (v2.2.0)
  - Locking symfony/polyfill-ctype (v1.26.0)
  - Locking symfony/swiftmailer-bundle (v2.2.6)
  - Locking symfony/symfony (v2.2.11)
  - Locking twig/extensions (v1.0.1)
  - Locking twig/twig (v1.44.6)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 10 installs, 25 updates, 0 removals
  - Downloading symfony/polyfill-ctype (v1.26.0)
  - Downloading twig/twig (v1.44.6)
  - Downloading psr/log (1.1.4)
  - Downloading psr/cache (3.0.0)
  - Downloading doctrine/lexer (1.2.3)
  - Downloading doctrine/annotations (1.13.2)
  - Downloading doctrine/reflection (1.2.3)
  - Downloading doctrine/event-manager (1.1.1)
  - Downloading doctrine/collections (1.6.8)
  - Downloading doctrine/cache (1.13.0)
  - Downloading doctrine/persistence (1.3.8)
  - Downloading doctrine/inflector (1.4.4)
  - Downloading doctrine/common (2.13.3)
  - Downloading symfony/symfony (v2.2.11)
  - Downloading jdorn/sql-formatter (v1.2.17)
  - Downloading doctrine/dbal (v2.4.5)
  - Downloading doctrine/doctrine-bundle (v1.2.0)
  - Downloading doctrine/orm (v2.4.8)
  - Downloading jms/cg (1.2.0)
  - Downloading phpoption/phpoption (1.8.1)
  - Downloading jms/parser-lib (1.0.1)
  - Downloading jms/metadata (1.7.0)
  - Downloading jms/aop-bundle (1.1.0)
  - Downloading jms/di-extra-bundle (1.3.0)
  - Downloading jms/security-extra-bundle (1.4.0)
  - Downloading sensio/distribution-bundle (v2.2.6)
  - Downloading sensio/framework-extra-bundle (v2.2.6)
  - Downloading sensio/generator-bundle (v2.2.6)
  - Downloading kriswallsmith/assetic (v1.1.3)
  - Downloading symfony/assetic-bundle (v2.1.3)
  - Downloading monolog/monolog (1.27.1)
  - Downloading symfony/monolog-bundle (v2.2.0)
  - Downloading swiftmailer/swiftmailer (v5.0.3)
  - Downloading symfony/swiftmailer-bundle (v2.2.6)
  - Downloading twig/extensions (v1.0.1)
  - Installing symfony/polyfill-ctype (v1.26.0): Extracting archive
  - Upgrading twig/twig (v1.12.2 => v1.44.6): Extracting archive
  - Upgrading psr/log (1.0.0 => 1.1.4): Extracting archive
  - Installing psr/cache (3.0.0): Extracting archive
  - Installing doctrine/lexer (1.2.3): Extracting archive
  - Installing doctrine/annotations (1.13.2): Extracting archive
  - Installing doctrine/reflection (1.2.3): Extracting archive
  - Installing doctrine/event-manager (1.1.1): Extracting archive
  - Installing doctrine/collections (1.6.8): Extracting archive
  - Installing doctrine/cache (1.13.0): Extracting archive
  - Installing doctrine/persistence (1.3.8): Extracting archive
  - Installing doctrine/inflector (1.4.4): Extracting archive
  - Upgrading doctrine/common (2.3.0 => 2.13.3): Extracting archive
  - Upgrading symfony/symfony (v2.2.0 => v2.2.11): Extracting archive
  - Upgrading jdorn/sql-formatter (v1.2.0 => v1.2.17): Extracting archive
  - Upgrading doctrine/dbal (2.3.2 => v2.4.5): Extracting archive
  - Upgrading doctrine/doctrine-bundle (v1.2.0-beta1 => v1.2.0): Extracting archive
  - Upgrading doctrine/orm (2.3.2 => v2.4.8): Extracting archive
  - Upgrading jms/cg (1.0.0 => 1.2.0): Extracting archive
  - Upgrading phpoption/phpoption (1.1.0 => 1.8.1): Extracting archive
  - Upgrading jms/parser-lib (1.0.0 => 1.0.1): Extracting archive
  - Upgrading jms/metadata (1.2.0-RC => 1.7.0): Extracting archive
  - Upgrading jms/aop-bundle (1.0.0 => 1.1.0): Extracting archive
  - Upgrading jms/di-extra-bundle (1.3.0 1.3.0 => 1.3.0 47e7c35): Extracting archive
  - Upgrading jms/security-extra-bundle (1.4.0 1.4.0 => 1.4.0 5eb806f): Extracting archive
  - Upgrading sensio/distribution-bundle (v2.2.0 => v2.2.6): Extracting archive
  - Upgrading sensio/framework-extra-bundle (v2.2.0 => v2.2.6): Extracting archive
  - Upgrading sensio/generator-bundle (v2.2.0 => v2.2.6): Extracting archive
  - Upgrading kriswallsmith/assetic (v1.1.0-alpha3 => v1.1.3): Extracting archive
  - Upgrading symfony/assetic-bundle (v2.1.2 => v2.1.3): Extracting archive
  - Upgrading monolog/monolog (1.4.0 => 1.27.1): Extracting archive
  - Upgrading symfony/monolog-bundle (v2.2.0 v2.2.0 => v2.2.0 b492800): Extracting archive
  - Upgrading swiftmailer/swiftmailer (v4.3.0 => v5.0.3): Extracting archive
  - Upgrading symfony/swiftmailer-bundle (v2.2.0 => v2.2.6): Extracting archive
  - Upgrading twig/extensions (v1.0.0-alpha => v1.0.1): Extracting archive
16 package suggestions were added by new dependencies, use `composer suggest` to see details.
Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Package sensio/distribution-bundle is abandoned, you should avoid using it. No replacement was suggested.
Package sensio/generator-bundle is abandoned, you should avoid using it. Use symfony/maker-bundle instead.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Package symfony/assetic-bundle is abandoned, you should avoid using it. Use symfony/webpack-encore-pack instead.
Package symfony/swiftmailer-bundle is abandoned, you should avoid using it. Use symfony/mailer instead.
Package twig/extensions is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
10 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap
PHP Fatal error:  Uncaught TypeError: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap(): Argument #1 ($event) must be of type Composer\Script\CommandEvent, Composer\Script\Event given, called in phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php on line 389 and defined in /home/clients/0f5bfc8311a1b196e18a753784946aa4/web/neorural/task_manager/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:32
Stack trace:
#0 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(389): Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap(Object(Composer\Script\Event))
#1 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(238): Composer\EventDispatcher\EventDispatcher->executeEventPhpScript('Sensio\\Bundle\\D...', 'buildBootstrap', Object(Composer\Script\Event))
#2 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(119): Composer\EventDispatcher\EventDispatcher->doDispatch(Object(Composer\Script\Event))
#3 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Installer.php(372): Composer\EventDispatcher\EventDispatcher->dispatchScript('post-update-cmd', true)
#4 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Command/UpdateCommand.php(248): Composer\Installer->run()
#5 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Command/Command.php(245): Composer\Command\UpdateCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(835): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(185): Symfony\Component\Console\Application->doRunCommand(Object(Composer\Command\UpdateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Console/Application.php(336): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(117): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Console/Application.php(131): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 phar:///opt/php8.0/bin/composer2.phar/bin/composer(83): Composer\Console\Application->run()
#12 /opt/php8.0/bin/composer2.phar(29): require('phar:///opt/php...')
#13 {main}
  thrown in /home/clients/0f5bfc8311a1b196e18a753784946aa4/web/neorural/task_manager/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php on line 32

Fatal error: Uncaught TypeError: Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap(): Argument #1 ($event) must be of type Composer\Script\CommandEvent, Composer\Script\Event given, called in phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php on line 389 and defined in /home/clients/0f5bfc8311a1b196e18a753784946aa4/web/neorural/task_manager/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php:32
Stack trace:
#0 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(389): Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::buildBootstrap(Object(Composer\Script\Event))
#1 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(238): Composer\EventDispatcher\EventDispatcher->executeEventPhpScript('Sensio\\Bundle\\D...', 'buildBootstrap', Object(Composer\Script\Event))
#2 phar:///opt/php8.0/bin/composer2.phar/src/Composer/EventDispatcher/EventDispatcher.php(119): Composer\EventDispatcher\EventDispatcher->doDispatch(Object(Composer\Script\Event))
#3 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Installer.php(372): Composer\EventDispatcher\EventDispatcher->dispatchScript('post-update-cmd', true)
#4 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Command/UpdateCommand.php(248): Composer\Installer->run()
#5 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Command/Command.php(245): Composer\Command\UpdateCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(835): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(185): Symfony\Component\Console\Application->doRunCommand(Object(Composer\Command\UpdateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Console/Application.php(336): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 phar:///opt/php8.0/bin/composer2.phar/vendor/symfony/console/Application.php(117): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 phar:///opt/php8.0/bin/composer2.phar/src/Composer/Console/Application.php(131): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 phar:///opt/php8.0/bin/composer2.phar/bin/composer(83): Composer\Console\Application->run()
#12 /opt/php8.0/bin/composer2.phar(29): require('phar:///opt/php...')
#13 {main}
  thrown in /home/clients/0f5bfc8311a1b196e18a753784946aa4/web/neorural/task_manager/vendor/sensio/distribution-bundle/Sensio/Bundle/DistributionBundle/Composer/ScriptHandler.php on line 32`

**Reply by ahundiak** [reply]
Put the composer.lock file back (actually just restore your backup completely) and then run just `composer update doctrine/orm`.  No options even if you think you need them.  If this does not work then well don't upgrade your PHP version.  There is zero chance of getting some sort of complete composer update to work.

**Comment by inolil** [comment]
I did restore the files and db.
I did ran `composer update doctrine/orm` and get this :
`The lock file does not contain require-dev information, run install with the --no-dev option or delete it and run composer update to generate a new lock file.  `
That's why I previously tried to remove the lock file.
PHP version is 7.3.

**Reply by stof** [reply]
oh, this looks like it is a lock file that was generated with a very old version of composer then, as it is been a _very_ long time that the lock file is not impacted by `--no-dev` anymore.

**Reply by ahundiak** [reply]
Yep.  OP - do a `composer --version`.  You definitely need a version 1.x to get anywhere with that lock file.  Maybe even less that one.

**Reply by inolil** [reply]
```
$ composer --version

Composer version 2.2.9 2022-03-15 22:13:37
```

I don't understand how a so recent date could appear here... I did a text comparison between the current project and the 25th June backup and everything matches except the content of the app/cache/prod folder.

The only files modified recently (June the 26) figures in the app/cache/prod and /app/logs folders.

No other files modified after 2019.

**Reply by ahundiak** [reply]
Composer is a package manager and not part of Symfony.  Somewhere along the line someone did a composer self-update and updated it.  Possibly your host provider.  So now you have the fun of removing the current version and installing a 1.x version.  Hopefully you are not using a shared host because most other developers are going to want to use the newer version.

Composer is a great little utility but someone was under the influence when they decided to drop backwards compatibility with composer.lock files.

**Reply by inolil** [reply]
OK. I contacted the technical support to see if there is a chance to run a composer1.x version.
I will probably need to install a dev env on my computer... Maybe the easiest way to get this application work again is past the code in a fresh install of current symfony and php versions and refactor all... But what about the database...
Regarding the log from my previous update attempt, doctrine and twig are no more supported :S

All advice are welcome.

Thanks again for your help

**Comment by ghost** [comment]
Just find a dev that is still in the business and pay him his hour income and he will set this up for you.
